### PR TITLE
remove dirty data in zfcp.conf

### PR DESCRIPTION
**Pull request by SharpRazor**
remove the dirty data in /etc/zfcp.conf left in image

Signed-off-by: bjcb <bjcb@cn.ibm.com>

### CODE REVIEW

**Comment by SharpRazor** [comment]
### test on rhel82
#### create a rhel82 bfv instance and attach a volume
the zfcp.conf content:
```
[root@rhel82-bfv ~]# cat /etc/zfcp.conf

0.0.5c07 0x5005076802400c1b 0x0001000000000000
0.0.5c07 0x5005076802400c1a 0x0001000000000000
0.0.5d0e 0x5005076802400c1b 0x0001000000000000
0.0.5d0e 0x5005076802400c1a 0x0001000000000000
[root@rhel82-bfv ~]#
```

#### capture to image

#### deploy and check the new zfcp.conf:
we can see that the content of  zfcp.conf was cleared:
```
[root@rhel82-capture-with-volume ~]# ll /etc/zfcp.conf
-rw-r--r--. 1 root root 1 Aug 18 02:45 /etc/zfcp.conf
[root@rhel82-capture-with-volume ~]# cat /etc/zfcp.conf

[root@rhel82-capture-with-volume ~]#
```
because we still not solve the purge reader problem, so the zfcp.conf do not have the new content.

**Comment by jackydalong** [comment]
1. Move the 3 lines here
image

because 
image

2. Change `Clear` to `Purge` for the 2nd line
@SharpRazor

**Comment by SharpRazor** [comment]
@jackydalong Hi, we need to know the OS version before purge zfcp.conf, because `sles` and `ubuntu` do not have zfcp.conf.
but the code of judging OS is  between line 805 and line 828, to avoid duplicated codes, so I wrote like this.

**Comment by jackydalong** [comment]
> @jackydalong Hi, we need to know the OS version before purge zfcp.conf, because `sles` and `ubuntu` do not have zfcp.conf.
> but the code of judging OS is between line 805 and line 828, to avoid duplicated codes, so I wrote like this.

Agree, though currently BFV or refresh_bootmap is NOT supported for SLES and Ubuntu.

**Comment by SharpRazor** [comment]
> > @jackydalong Hi, we need to know the OS version before purge zfcp.conf, because `sles` and `ubuntu` do not have zfcp.conf.
> > but the code of judging OS is between line 805 and line 828, to avoid duplicated codes, so I wrote like this.
> 
> Agree, though currently BFV or refresh_bootmap is NOT supported for SLES and Ubuntu.
> @SharpRazor
> ### Just a reminder : change `Clear` to `Purge` or `Clean` for the 2nd line
> 
> @SharpRazor

thanks, done.

**Review by jackydalong — APPROVED** [review]
LGTM

File: zthin-parts/zthin/bin/refresh_bootmap [hunk 1]
```diff
@@ -843,6 +843,9 @@ function refreshZIPL {
       printError "Failed to execute zipl on $os due to ${deviceMountPoint}/${zipl_conf} not exist."
       exit 13
     else
+      # Remove dirty data in /etc/zfcp.conf left in image
+      inform "Purge contents in ${deviceMountPoint}/etc/zfcp.conf"
```
Comment (dongyanyang) [inline zthin-parts/zthin/bin/refresh_bootmap:847]: 1. move these change to after line 802. where we updated the fstab and cleared multipath bindings/wwids.
2. better to check whether the zfcp.conf file exists and writable with a "-r"

**Review by dongyanyang — CHANGES_REQUESTED** [review]

**Comment by dongyanyang** [comment]
> @jackydalong Hi, we need to know the OS version before purge zfcp.conf, because `sles` and `ubuntu` do not have zfcp.conf.
> but the code of judging OS is between line 805 and line 828, to avoid duplicated codes, so I wrote like this.

just saw Jacky and your comments.
Your concern can be solved by adding a [[ if -r xxx ]] before your code. Only do the change when the file exists and readable.

**Comment by SharpRazor** [comment]
@dongyanyang thanks, I have changed to use if statements.

**Review by dongyanyang — APPROVED** [review]
lgtm

**Pull request merged**